src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPFoldingRangeFixtureTestCase.java

```diff
@@ -90,6 +120,9 @@ protected void assertFoldingRanges(@NotNull String fileName,
             String expectedPlaceholderText = StringUtil.isNotEmpty(mockCollapsedText) ? mockCollapsedText : "...";
             String actualPlaceholderText = actualFoldRegion.getPlaceholderText();
             assertEquals(expectedPlaceholderText, actualPlaceholderText);
+
+            // Check the initial expansion state
+            assertEquals("Incorrect expansion state for region " + (i + 1) + ".", ArrayUtil.contains(i + 1, collapsedByDefaultRangeNumbers), !actualFoldRegion.isExpanded());
```

Comment:
This is where it checks whether each fold region is in the correct initial expanded vs. collapsed state.